src/drivers/imx/sai.c

```diff
@@ -366,7 +375,12 @@ static inline int sai_set_config(struct dai *dai, struct ipc_config_dai *common_
 		break;
 	}
 
-	val_cr4 |= REG_SAI_CR4_FRSZ(sai->params.tdm_slots);
+	if (tdm_mode == TDM_ENABLE)
+		val_cr4 |= REG_SAI_CR4_FRSZ(sai->params.tdm_slots);
+	else
+		val_cr4 |= REG_SAI_CR4_FRSZ(
+		    (sai->params.tdm_slots == 1) ? 2 : sai->params.tdm_slots);
+
 	val_cr4 |= REG_SAI_CR4_CHMOD;
```

Comment:
It seems conflict with the variable tdm_mode, because we always set mode OUTPUT mode here?

Reply:
I don't think TDM mode from my patch is the same as TDM mode configuration of output mode.

TDM mode says that transmit data pins are tri-stated when slots are masked or channels disabled. You can have I2S mode enabled and output mode set to tdm.